src/mender-update/deployments/platform/boost_log/deployments.cpp

```diff
@@ -81,10 +81,14 @@ error::Error DeploymentLog::PrepareLogDirectory() {
 
 error::Error DeploymentLog::DoPrepareLogDirectory() {
 	fs::path dir_path(data_store_dir_);
-	bool created = fs::create_directories(dir_path);
-	if (created) {
-		// should basically never happen, but if we happened to create the
-		// directory, it's empty and thus well-prepared
+
+	// `create_directories` may fail on some platforms with an error if the destination exists
+	// as a symlink, even if the target is a valid directory. So let's check for existence
+	// first.
+	if (not fs::exists(dir_path)) {
```

Comment:
I'm so proud I'm in tears :smiling_face_with_tear:

Reply:
Hehe, the `not` is nice!

Reply:
I have to confess that I started using it also (as per yesterday on my local development). It is amazing how the `not` `not` nice nitpicks spread.